ENHANCE: Option `NoPrecomputedData` to avoid use of data libraries

hulpke opened the pull request:
### Tick all what applies to this pull request

- [x] Adds new features
- [x] Improves and extends functionality

### Write below the description of changes (for the release notes)
Add the function option `NoPrecomputedData` to turn off reliance on data libraries. This applies for all uses of Small groups/Tom/Prim/Trans data libraries in the main GAP library.

Added deocumentation in section on data libraries.

Commit:
ENHANCE: Option `NoPrecomputedData` to avoid use of data libraries

Commit:
ENHANCE: Warning/disable use of perfect groups library in algorithm

Commit:
Test for NoPrecomputedData

Commit:
Tests for use of small groups library / IdGroup

fingolfin commented:
Seems plausible to me.

Commit:
Enabled test for maximal primitive from library, using data library flag.

hungaborhorvath commented:
Some of these commits are already in master. Maybe an accidental merge happened instead of a rebase?

hulpke commented:
@hungaborhorvath 
I don't think so -- what seems to be rather happening is that a rebase included several commits in master again as part of this branch. I'll see to fix this.

Commit:
Test for data library use flag

Commit:
Added documentation of data library use for calculations

and how to check/turn it off. This might need to be moved to a more
prominent place.
